### ISSUE TITLE
`SupportsNext[Any]*` typechecks to be non-iterable when used inside methods

### ISSUE THREAD

**Issue by Kache** [microsoft/pyright#3682]
**Describe the bug**
`SupportsNext[Any]*` typechecks to be non-iterable when used inside methods.

**To Reproduce**
See below

**Expected behavior**
`SupportsNext[Any]*` should typecheck to be iterable when used inside methods.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.

```python
def yield_from(foo):
    iter_foo = iter(foo)
    yield from iter_foo

def for_in(foo):
    iter_foo = iter(foo)
    for x in iter_foo:
        yield x


class Foo:
    def __init__(self, foo):
        self._foo = iter(foo)

    def yield_from(self):
        yield from self._foo

    def for_in(self):
        for x in self._foo:
            yield x
```

<img width="821" alt="image" src="https://user-images.githubusercontent.com/133218/178128125-41b76c51-783b-4857-82ef-ad42c6b4549e.png">

**VS Code extension or command-line**
Vim with coc.nvim + coc-pyright: `[coc.nvim] coc-pyright 1.1.232 with Pyright 1.1.235`

**Additional context**

**Comment by erictraut** [comment]
I'm not able to repro the problem.

You're using pyright 1.235 which is 23 versions (!!!) out of date. Please upgrade to the latest version.

**Comment by Kache** [comment]
Ah, sorry about that. Upgrading has indeed resolved the issue.
